File: multify/lib/multify/task.rb
class Multify::Task < Multify::Resource

  extend Multify::Resource::WithSlug

  assignable_attributes :name, :slug, :project_id, :done, :due_at

  attribute :name,       String
  attribute :slug,       String
  attribute :project_id, Integer
  attribute :done,       Boolean
  attribute :due_at,     Time
  attribute :created_at, Time
  attribute :updated_at, Time

  def project
    @project ||= Multify::Project.find(id: project_id) if project_id
  end

  def project= project
    @project = project
    project_id = project.id
  end

end

File: www_static/javascripts/multify/task.js
Multify.Task = Backbone.Model.extend({

  initialize: function(attrs, options){
    this.project(options.collection.project);
  },

  id: null,
  project_id: null,
  name: null,
  slug: null,
  created_at: null,
  updated_at: null,

  project: function(project){
    if (project instanceof Backbone.Model){
      this._project = project;
      this.set('project_id', project.id);
    }
    return this._project;
  }
});

Multify.Task.modelName = 'task';
Multify.Task.path = 'tasks';


